RELEASE_PROCEDURE.md

```diff
@@ -25,14 +25,30 @@ DLA-Future follows [Semantic Versioning](https://semver.org).
 1. Add a link to the documentation for the release in `DOCUMENTATION.md` and update the link in `README.md`.
    The documentation will be generated automatically after the `vx.y.z` tag has been created and pushed.
 
+1. Merge all the changes to the `master` branch (also for patch releases).
+
 1. Ensure you have `gh` ([GitHub CLI](https://cli.github.com)) installed. Run `gh auth login` to authenticate
    with your GitHub account, or set `GITHUB_TOKEN` to a token with `public_repo` access.
 
+1. Ensure you are working from the `eth-cscs/DLA-Future` and not your own fork for following steps.
+
+1. For patch releases: switch to `version_X.Y` branch.
+
+1. For patch releases: cherry pick the commits for the patch from `master` to the `version_X.Y` branch.
+   Make sure the cherry-picked commits are listed in the `CHANGELOG.md`, and make sure you cherry-picked
+   the PR(s) updating all the above (version, documentation, etc.).
+
+1. Check the release branch to make sure the content matches the release notes.
+
 1. Create a release on GitHub using the script `scripts/roll_release.sh`. This
    script automatically tags the release with the corresponding release number.
 
+1. For patch releases: merge the `version_X.Y` branch into `master`.
+
 1. Update spack recipe in `spack/packages/dla-future/package.py` adding the new release.
 
+1. For patch releases: update `CHANGELOG.md` in `master` to match the content of the patch release.
+
```

Comment:
I guess these two `For patch releases:` points may not be necessary with the other points about always cherry pikcing from `master`? Not sure if I'm missing something, but in principle all changes should already be on the `master` branch. On the other hand, I also don't mind keeping the steps in case we have to make some manual changes on the release branch?

Reply:
Yes, I noticed it too this morning that after adding what Raffaele suggested (to always push things on `master`), this is no longer necessary.